src/ThresholdNetworkUpgradeable.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT AND Apache-2.0
+pragma solidity ^0.8.19;
+
+import {Initializable} from
+    "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {IThresholdNetwork} from "./interfaces/IThresholdNetwork.sol";
+import {G1Point} from "./utils/Bn128.sol";
+
+/// @title ThresholdNetwork
+/// @author Cryptonet
+/// @notice This contract represents a threshold network.
+/// @dev All threshold networks have a distributed key;
+/// the DKG contract facilitates the generation of a key, whereas Oracle contracts are given a key
```

Comment:
Update comments

Reply:
And what's the difference between this one and ThresholdNetwork ? Why keep both ?

Reply:
ThresholdNetworkUpgradeable uses `initialize()` and ThresholdNetwork uses `constructor()` (this is really the only key difference that makes a contract compatible with upgradeablitity).

I kept ThresholdNetwork to use in DKG since DKG is not upgradeable, but technically we can just use a single `ThresholdNetwork` that uses `initialize()`, which makes it compatible with both upgradeable and non-upgradeable contracts.

I thought making the distinction would make it more obvious for people reading the code that DKG is not upgradeable.

Reply:
Yeah I'm a bit confused still - we should only use one if possible. 
* @jaeaster why is the DKG not upgradeable ? It's not _strictly_ needed now, but i thought we wanted it to be upgradeable so we can have resharing inplace without starting a new network. Not a big deal  / mandatory because for mainnet v0 we control them anyway but just wanted to clarify that point.
* Maybe we don't really need to have a `ThresholdNetwork` for the DKG, do we ?

Reply:
Yeah I forgot about resharing for the DKG. We can make the DKG upgradeable OR we can do resharing in another contract. In another contract, we would likely need to copy over some state.